pandas/core/dtypes/common.py

```diff
@@ -28,6 +28,7 @@
     is_array_like,
     is_bool,
     is_complex,
+    is_dataclass_instance,
```

Comment:
just call is_dataclass

Reply:
a dataclass is a class that is defined with `@dataclass`, I think that `is_dataclass` would imply that we are checking if a class is a dataclass, while this name implies that what I am looking at should be a dataclass instance (not I even have the `not isinstance(item, type)` in the check)

Reply:
this is not correct these are all type inferences routines of instances so adding _instance is redundant and not conforming with the style
and is_dataclass is appropriate

Reply:
ok

Reply:
sorry to circle back on this, but I believe that I am quite right here, given how the documentation refers to dataclasses https://docs.python.org/3/library/dataclasses.html#dataclasses.is_dataclass

The documentation clearly lists a distinction when it says "Return True if its parameter is a dataclass or an instance of one"

The dataclass decorator and the make_dataclass function both return a dataclass object from which dataclass instances can be made.

If I declared

```python
Point = make_dataclass("Point", [("x", int), ("y", int)])
p = Point(0,1)
```

Then `Point` is the dataclass and `p`  is an instance of the dataclass `Point`.

This check looks to see if its the latter and not the former.